lib/core-classes/PayloadService.ts

```diff
@@ -151,45 +151,61 @@ export class PayloadService {
       throw new BadRequestError(`Device ${device._id} is not allowed for registration.`);
     }
 
-    const ret = await this.register(device, decoder, request, { refresh });
+    const hasContent = catalogEntry && catalogEntry.content && catalogEntry.content.tenantId;
+    const tenantCatalogEntry = hasContent ? await this.getCatalogEntry(catalogEntry.content.tenantId, device._id) : undefined;
+
+    const response = await global.app.trigger('device-manager:device:provisioning:before', {
+      device,
+      adminCatalog: catalogEntry,
+      tenantCatalog: tenantCatalogEntry,
+    });
+
+    const enrichedDevice = response.device ? response.device : device;
```

Comment:
If the pipe does not returns and object with the same content it's normal if it crashes. In the core we don't use this kind of silent substitution